sound/soc/sof/intel/hda-ipc.c

```diff
@@ -62,11 +62,9 @@ int hda_dsp_is_ipc_ready(struct snd_sof_dev *sdev)
 	/* is DSP ready for next IPC command */
 	busy = snd_sof_dsp_read(sdev, HDA_DSP_BAR, HDA_DSP_REG_HIPCI);
 	done = snd_sof_dsp_read(sdev, HDA_DSP_BAR, HDA_DSP_REG_HIPCIE);
-	if ((busy & HDA_DSP_REG_HIPCI_BUSY) ||
-	    (done & HDA_DSP_REG_HIPCIE_DONE))
-		return 0;
 
-	return 1;
+	return !(busy & HDA_DSP_REG_HIPCI_BUSY) &&
+		!(done & HDA_DSP_REG_HIPCIE_DONE);
```

Comment:
This commit is not needed right? You ended up removing the API after all

Reply:
right, thanks for spotting, squashed.